lib/matplotlib/cbook.py

```diff
@@ -361,9 +361,11 @@ class _BoundMethodProxy(object):
     Minor bugfixes by Michael Droettboom
     '''
     def __init__(self, cb):
+        self._hash = hash(cb)
+        self._callbacks = []
```

Comment:
Can we go back to the beginning for a moment? What exactly is the bug being
fixed here? I have added bound methods before to the callback registry
before without any (apparent) issues. It might help test-making if we can
cleanly describe a situation where the existing code is buggy.

On Thu, Feb 19, 2015 at 8:44 AM, OceanWolf <notifications@github.com> wrote:

> In lib/matplotlib/cbook.py
> <https://github.com/matplotlib/matplotlib/pull/4118#discussion_r24987757>:
>
> > @@ -361,9 +361,11 @@ class _BoundMethodProxy(object):
> >      Minor bugfixes by Michael Droettboom
> >      '''
> >      def __init__(self, cb):
> > +        self._hash = hash(cb)
> > +        self._callbacks = []
>
> Almost, they get called when the object (of the method we proxy), gets
> destroyed, if we proxy a object method, otherwise it will never get called
> (then again I don't think a class method nor a function would get
> destroyed).
>
> So basically I use this as a way to tell those that hardref'ed to this
> _BoundMethodProxy "I have died inside, so let me go so that I myself can
> die (and get GC'ed)"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4118/files#r24987757>.
>